### PR TITLE
iOS - InterApp - Open app fallback to open a url

### PR DESCRIPTION
**Pull request by AndrewEQ**
Check if app is available, else can open a link (typically to App Store).

For the app that you want to open, you have to know its scheme and whitelist it:

Example:
```
whatsapp://
```
```
<Extensions Backend="CPlusPlus" Condition="iOS">
  <Require Condition="iOS" Xcode.Plist.Element>
    <![CDATA[
      <key>LSApplicationQueriesSchemes</key>
      <array>
        <string>whatsapp</string>
      </array>
    ]]>
  </Require>
</Extensions>
```

### CODE REVIEW

File: Source/Fuse.Launcher.InterApp/InterApp/JS.uno [hunk 1]
```diff
@@ -232,7 +232,14 @@ namespace Fuse.Reactive.FuseJS
 		*/
 		public static object LaunchApp(Scripting.Context context, object[] args)
 		{
-			Fuse.LauncherImpl.InterAppLauncher.LaunchApp((string)args[0]);
+			if defined(Android)
+			{
+				Fuse.LauncherImpl.InterAppLauncher.LaunchApp((string)args[0],"");
```
Comment (mortend) [inline Source/Fuse.Launcher.InterApp/InterApp/JS.uno]: Code-style nit: Add space after comma in argument list.

File: Source/Fuse.Launcher.InterApp/InterApp/JS.uno [hunk 2]
```diff
@@ -232,7 +232,14 @@ namespace Fuse.Reactive.FuseJS
...
+			}
+			if defined(iOS)
+			{
+				Fuse.LauncherImpl.InterAppLauncher.LaunchApp((string)args[0],(string)args[1]);
```
Comment (mortend) [inline Source/Fuse.Launcher.InterApp/InterApp/JS.uno]: Code-style nit: Add space after comma in argument list.

File: Source/Fuse.Launcher.InterApp/InterApp/InterAppLauncher.uno [hunk 3]
```diff
@@ -40,7 +40,7 @@ namespace Fuse.LauncherImpl
 			}
 		}
 
-		public static void LaunchApp(string uri)
+		public static void LaunchApp(string uri, string appStoreUri)
```
Comment (mortend) [inline Source/Fuse.Launcher.InterApp/InterApp/InterAppLauncher.uno]: Can the `appStoreUri` parameter be optional, as in `string appStoreUri = null`?

Comment (AndrewEQ) [reply Source/Fuse.Launcher.InterApp/InterApp/InterAppLauncher.uno]: Sure can, wasn't sure we could do that in Uno, when's the Uno Handbook coming out? 😄

Comment (mortend) [reply Source/Fuse.Launcher.InterApp/InterApp/InterAppLauncher.uno]: Haha :D

**Comment by AndrewEQ** [comment]
Sorted.

File: Source/Fuse.iOS/iOSInterop.uno [hunk 4]
```diff
@@ -56,5 +56,22 @@ namespace Fuse.iOS.Bindings
...
+				dispatch_sync(dispatch_get_main_queue(), ^{
+					[[UIApplication sharedApplication] openURL:[NSURL URLWithString:uri]];
+				});
+			} 
+			else 
+			{
+				dispatch_sync(dispatch_get_main_queue(), ^{
+					[[UIApplication sharedApplication] openURL:[NSURL URLWithString:appStoreUri]];
+				});
+			}
```
Comment (mortend) [inline Source/Fuse.iOS/iOSInterop.uno]: Do we need to check that `else if (appStoreUri != null && appStoreUri.length() > 0)` here? (This is pseudo-code and probably not valid Obj-C.)

Comment (AndrewEQ) [reply Source/Fuse.iOS/iOSInterop.uno]: sorted.

**Pull request merged**